pkg/generate/app/sourcelookup.go

```diff
@@ -467,13 +470,17 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		source.DockerfileContents = repo.Info().Dockerfile.Contents()
 	}
 	if !repo.ignoreRepository {
-		remoteURL, err := repo.RemoteURL()
+		remoteURL, ok, err := repo.RemoteURL()
 		if err != nil {
 			return nil, nil, fmt.Errorf("cannot obtain remote URL for repository at %s", repo.location)
 		}
-		source.URL = remoteURL
-		source.Ref = remoteURL.Fragment
-		source.ContextDir = repo.ContextDir()
+		if ok {
+			source.URL = remoteURL
+			source.Ref = remoteURL.Fragment
+			source.ContextDir = repo.ContextDir()
+		} else {
+			source.Binary = true
```

Comment:
Yup, missed this. I do agree. If I give you a context dir I would expect it to be honored even if I'm only going to do binary builds.